### DISCUSSION TITLE
Use Filament as the actual application

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2198] [Help] [answered]
Hi guys,

I know that Filament is rather meant to work on top of another (laravel based) application.
Still, I'd like to have Filament with its design and functions (+ a bit other stuff/pages) to be the actual application.

Is there a way to use it that way? I was thinking of the following stack:
- Laravel
- Jetstream
- Livewire
- Filament

So, after login user will be redirected to the Filament dashboard and can use all the Jetstream functions from it (using the function within the Filament layout.

Appreciate any remark on that.
Thank you, have a nice weekend.

**Comment by danharrin** [comment, marked as answer]
Hey, I know of many people who use Filament as the entire app. We have a few plugins that enable auth + profile features in just one command, like https://github.com/jeffgreco13/filament-breezy

Then you just set Filament's `path` in config to be `''`, and you have the entire app built!

**Reply by vv-ac** [reply]
Hi Dan, thanks for replying that fast!
I saw your posts on reddit earlier, so I took the peek into the docs (how great is that package???).

Having breezy means I get rid of Jetstream, right? Besides MFA and the (personally) not needed "Teams" function, there is no difference then, correct?

Thank you for your great work on Filament - looks so promising.

**Reply by danharrin** [reply]
Yeah it replaces Jetstream with a Breeze-like interface. And I believe that the Breezy author is working on adding 2FA, right @jeffgreco13?

**Reply by jeffgreco13** [reply]
@danharrin that's correct! I've been working on it, and hope to have a release with 2FA tagged by the end of this weekend.

@riskmate-ac to your point, you're absolutely correct that Filament + Breezy should replace the need for Jetstream. I now have 2 fully managed applications where the entire UI is just Filament. I use Breezy for authentication flow, then use Shield to control who has access to what.

**Reply by vv-ac** [reply]
Thank you for taking the time! Nice to hear, that MFA will come along with breezy, nothing to wish for on top of that.

I just did a fresh install, laravel/laravel, livewire, filament and breezy. Then I changed the config/filament.php to point to the Breezy Login::class.
Login works, all good. I then altered config/filament.php to:
```
'path' => env('FILAMENT_PATH', ''),
```
Also in web.php:
```
Route::get('/', function () {
    return view('filament.auth.login');
});
```

When I now access the root it returns:
![grafik](https://user-images.githubusercontent.com/67121752/164911705-9f1cfbbc-4c94-4ea6-92a5-10c52231e643.png)

When I run route:list it returns that it actually exists:
![grafik](https://user-images.githubusercontent.com/67121752/164911738-5baad8e0-9adf-44bc-8e70-2b4038b41ec4.png)

Clearing the routes did not help so far. I likely miss sth very obvious, don't I?

Thank you guys, have a nice Saturday.

**Reply by jeffgreco13** [reply]
Change your route to a redirect instead of view.

https://laravel.com/docs/9.x/redirects#redirecting-named-routes

You have the name of the route correct, that's just not the right location of the login view.

**Reply by vv-ac** [reply]
Cheers, sth obvious it was :-).
Thank you!

**Reply by DhrutiPandya0267** [reply]
hey @ristmate-ac 
i have a same problem
what did you add in route?
can you help me?

**Reply by jeffgreco13** [reply]
@DhrutiPandya0267 try this in web.php.

```php

Route::get('/', function () {
    return redirect()->route('filament.auth.login');
});

```

**Comment by vv-ac** [comment]
Not even trying to ask this one as you last solution did the trick.

Now, when I register a user and try to login the app wants to take me to filament.pages.dashboard but returns this:
![grafik](https://user-images.githubusercontent.com/67121752/164944814-4705ff93-2b4b-414e-b3da-f3fc3335d505.png)
![grafik](https://user-images.githubusercontent.com/67121752/164944821-cbe27f7e-c4ef-4683-ad7d-a87760ce68d5.png)

Even if I register the route in my web.php it does not work.
The page/blade file is there:
![grafik](https://user-images.githubusercontent.com/67121752/164944858-4e43191d-02c1-4b1c-85ac-155835938858.png)

Do you have another hint for me how to fix such a basic problem I encounter?

Thanks.

**Reply by vv-ac** [reply]
Just tested it once more by not installing breezy.
I did:
- Installaing new laravel/laravel
- Installing new livewire
- Installing new filament/filament
- Creating a filament-user
- Logging in works just fine
- Setting  'path' => env('FILAMENT_PATH', ''),
- Calling 127.0.0.1 and logging into Filament
- Getting the error:
![grafik](https://user-images.githubusercontent.com/67121752/164964671-e35271e6-f308-4f73-98f6-f52c75922abc.png)

Manually registering the route as a redirect in web.php does not help.

**Reply by vv-ac** [reply]
Ashes on my head, I simply did not publish the filament config.

**Reply by danharrin** [reply]
Ah so you just published part of the config and not all of it?

**Reply by vv-ac** [reply]
Yes, wrong timing and not looking closely enough - happens.

Now, that I had the chance to gather more experience with filament --> What a great set of functionalities to work with. You can create simple applications in less than an hour. Pretty cool stuff, so keep up the good work!

**Reply by danharrin** [reply]
Nice, I haven't found anything you realistically *can't* do with Filament yet!
